boards/Kconfig

```diff
@@ -19,6 +19,17 @@ config BOARD_REVISION
 	  "plank@foo", this option will be "foo". If BOARD is "plank",
 	  this option will be the empty string.
 
+config BOARD_TARGET
+	string
+	default "$(BOARD)@$(BOARD_REVISION)$(BOARD_IDENTIFIER)" if "$(BOARD_REVISION)" != ""
```

Comment:
It is confusing that we are still using the BOARD identifier terminology here.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/180017/37b9cd60-f621-4012-9f99-4e24e0f0cc2a)

Reply:
Agreed.

Reply:
maybe deprecate `CONFIG_BOARD` and introduce `CONFIG_BOARD_NAME`?

Reply:
They will be updated in a separate PR since there's usage through the whole zephyr tree of it as per https://github.com/zephyrproject-rtos/zephyr/pull/69904#issuecomment-2003607242

Reply:
I agree, we should fix this even though I still haven't finished the work documenting this in https://github.com/zephyrproject-rtos/zephyr/pull/69802

Reply:
more confusion with links to multiple PRs, all of which talk about different terminology, even the documentation change one still talks about identifiers. Can we do this in order somehow, right now it is impossible to know what is an oversight and what is there just for backward compatibility etc and what the next steps are.

Reply:
created https://github.com/zephyrproject-rtos/zephyr/pull/70438.

If we merge that one first, then we can rebase all PRs accordingly and use the new terminology everywhere.